federation/cluster/common.sh

```diff
@@ -24,6 +24,22 @@
 
 : "${KUBE_ROOT?Must set KUBE_ROOT env var}"
 
+# Provides the $KUBERNETES_PROVIDER variable and detect-project function
+source "${KUBE_ROOT}/cluster/kube-util.sh"
+
+# If $FEDERATION_PUSH_REPO_BASE isn't set set the GCR registry name based on the
+# detected project name.
+if [[ -z "${FEDERATION_PUSH_REPO_BASE}" ]]; then
```

Comment:
On second thought, should we also check that KUBERNETES_PROVIDER = gce or gke?
I assume this wont work on vagrant and aws?